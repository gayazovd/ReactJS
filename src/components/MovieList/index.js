import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import MovieListView from './view';
import NotFoundFilms from '../NotFoundFilms';

import classNames from 'classNames/bind';
import s from './style.scss';
const cx = classNames.bind(s);

class MovieList extends PureComponent {
    handleClickOnFilm = item => {
        this.props.getFilm(item);
    };

    renderFilms = movies => {
        if (movies.length > 0) {
            return <MovieListView onHandleClick={this.handleClickOnFilm} data={movies} />;
        }
        return <NotFoundFilms />;
    };

    render() {
        const { movies } = this.props;
        return <div className={cx('main-wrapper')}>{this.renderFilms(movies)}</div>;
    }
}
MovieList.propTypes = {
    movies: PropTypes.arrayOf(PropTypes.object).isRequired
};

const mapStateToProps = state => {
    return {
        movies: state.loadingMovies.movies
    };
};

export default withRouter(connect(mapStateToProps)(MovieList));
