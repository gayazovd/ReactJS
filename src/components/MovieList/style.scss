.main-wrapper{
    display: flex;
    justify-content: center;
    flex: 1 0 auto;
}
.wrapper{
    display: flex;
    padding: 10px;
    width: 870px;
    flex-wrap: wrap;
    justify-content: space-between;
}