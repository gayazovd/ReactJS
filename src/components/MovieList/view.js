import React from 'react';
import FilmPreview from '../FilmPreview';
import s from './style.scss';

import classNames from 'classNames/bind';
const cx = classNames.bind(s);

const MovieListView = ({ onHandleClick, data }) => (
  <div className={cx('wrapper')}>
    {data &&
      data.map(item => (
        <FilmPreview onClick={() => onHandleClick(item.id)} key={item.id} film={item} />
      ))}
  </div>
);

export default MovieListView;
