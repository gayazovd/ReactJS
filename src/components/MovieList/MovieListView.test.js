import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MovieListView from './view';
import MovieList from './index';

configure({ adapter: new Adapter() });

describe('MovieListView is render', () => {
  const onHandleClick = jest.fn();
  const props = {
    data: [{ id: 1 }, { id: 2 }],
    onHandleClick
  };

  const movieView = shallow(<MovieListView {...props} />);

  it('get snapshot component', () => {
    expect(movieView).toMatchSnapshot();
  });

  it('component <MovieListView/> is render', () => {
    const element = movieView.find('FilmPreview').at(1);
    element.simulate('click');
    expect(onHandleClick).toHaveBeenCalled();
  });
});
