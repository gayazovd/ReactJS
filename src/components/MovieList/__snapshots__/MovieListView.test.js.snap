// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListView is render get snapshot component 1`] = `
<div
  className="wrapper"
>
  <withRouter()
    film={
      Object {
        "id": 1,
      }
    }
    key="1"
  />
  <withRouter()
    film={
      Object {
        "id": 2,
      }
    }
    key="2"
  />
</div>
`;
